pppoatm: Allow wildcard ATM devices

When operating pppd's pppoatm plugin with an USB ADSL modem, e.g. an
Alcatel Speedtouch, the ATM device number might change when the modem is
reconnected to the USB port or when the host controller resets the USB
device.

This patch allows to specify the ATM device as wildcard which gives
enough flexibility to cope with changing device names.

The patch originated from the Debain project.

Signed-off-by: Jo-Philipp Wich <jo@mein.io>

--- a/pppd/plugins/pppoatm/pppoatm.c
+++ b/pppd/plugins/pppoatm/pppoatm.c
@@ -75,7 +75,7 @@ static int setdevname_pppoatm(const char
 	//info("PPPoATM setdevname_pppoatm: '%s'", cp);
 	memset(&addr, 0, sizeof addr);
 	if (text2atm(cp, (struct sockaddr *) &addr, sizeof(addr),
-	    T2A_PVC | T2A_NAME) < 0) {
+	    T2A_PVC | T2A_NAME | T2A_WILDCARD) < 0) {
                if(doit)
                    info("atm does not recognize: %s", cp);
 		return 0;
